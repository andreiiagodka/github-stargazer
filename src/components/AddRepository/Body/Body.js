import React from 'react'

import { Container, Col, Form, Button } from 'react-bootstrap'
import { useFormik } from 'formik'
import * as Yup from 'yup'

import { getRepositoryData } from '../../../shared/github'
import { createRepository } from '../../../shared/firebase'

const Body = props => {
  const initialValues = { name: '' }
  const validationSchema = Yup.object({
    name: Yup.string().required('Required')
  })

  const formik = useFormik({
    initialValues: initialValues,
    validationSchema: validationSchema,
    onSubmit: (values, { setSubmitting, setFieldError }) => {
      setSubmitting(true)
      getRepositoryData(values.name)
        .then(response => {
          const [ repository, languages ] = response

          createRepository(repository.data, languages.data)
          props.history.push('/')
        })
        .catch(() => {
          setFieldError('name', 'Repository not found')
        })
      setSubmitting(false)
    }
  })

  return (
    <Container fluid>
      <Col md={{ span: 8, offset: 2 }}>
        <Form onSubmit={formik.handleSubmit}>
          <Form.Group>
            <Form.Control
              id='name'
              name='name'
              placeholder='Enter repository name, e.g. rubygarage/truemail'
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              value={formik.values.name}
            />
            { formik.touched.name && formik.errors.name ? (<Form.Text className="text-danger">{formik.errors.name}</Form.Text>) : null }
          </Form.Group>
          <Button type='submit' variant='primary' size='lg' block>
            Add
          </Button>
        </Form>
      </Col>
    </Container>
  )
}

export default Body