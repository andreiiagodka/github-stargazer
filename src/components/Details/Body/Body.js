import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import { Row, Container } from 'react-bootstrap'

const Body = ({ repository }) => {
  const mappedLanguages = Object.keys(repository.languages).map(key => {
    return <li key={key}>{key}: {repository.languages[key]}</li>
  })

  return (
    <Fragment>
      <Container>
        <Row>
          <h3>Stats</h3>
        </Row>
        <Row>
          <ul style={{ listStyleType: "none" }}>
            <li>Stars: {repository.stargazers_count}</li>
            <li>Watchers: {repository.watchers_count}</li>
            <li>Forks: {repository.forks_count}</li>
          </ul>
        </Row>
      </Container>
      <Container>
        <Row>
          <h3>Languages</h3>
        </Row>
        <Row>
          <ul style={{ listStyleType: "none" }}>
            {mappedLanguages}
          </ul>
        </Row>
      </Container>
    </Fragment>
  )
}

Body.propTypes = {
  repository: PropTypes.object.isRequired
}

export default Body