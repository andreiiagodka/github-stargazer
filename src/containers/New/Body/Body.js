import React, { Component } from "react";
import {
  Container,
  Col,
  Card,
  Form,
  Button
} from "react-bootstrap";
import { Redirect } from 'react-router';
import { connect } from 'react-redux';

import * as actions from '../../../store/actions/actions';

class Body extends Component {
  submitHandler = event => {
    event.preventDefault()

    this.props.searchRepository(event.target.elements.name.value)
  }

  render() {
    let body = <Redirect to='/' />
    if (!this.props.success) {
      body = (
        <Card.Body>
          <Container fluid>
            <Col md={{ span: 8, offset: 2 }}>
              <Form onSubmit={this.submitHandler}>
                <Form.Group>
                  <Form.Control
                    id='name'
                    name='name'
                    placeholder='Enter repository name, e.g. rubygarage/truemail'
                  />
                  <Form.Text className="text-danger">{this.props.error}</Form.Text>
                </Form.Group>
                <Button type='submit' variant='primary' size='lg' block>
                  Add
                </Button>
              </Form>
            </Col>
          </Container>        
        </Card.Body>
      )
    }

    return body
  }
}

const mapStateToProps = state => {
  return {
    success: state.new.success,
    error: state.new.error
  }
}

const mapDispatchToProps = dispatch => {
  return {
    searchRepository: (name) => dispatch(actions.searchRepository(name))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Body)