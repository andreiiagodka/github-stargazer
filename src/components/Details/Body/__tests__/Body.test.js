import React from 'react'
import renderer from 'react-test-renderer'

import Body from '../Body'

describe('<Body /> component', () => {
  it('renders component correctly', () => {
    const props = {
      repository: {
        stargazers_count: 1,
        watchers_count: 2,
        forks_count: 3,
        languages: { 'Ruby': 4 }
      }
    }
    const tree = renderer.create(<Body {...props} />).toJSON()
  
    expect(tree).toMatchSnapshot()
  })
})
