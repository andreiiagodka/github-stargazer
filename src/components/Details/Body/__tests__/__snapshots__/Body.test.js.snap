// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Body /> component renders component correctly 1`] = `
Array [
  <div
    className="container"
  >
    <div
      className="row"
    >
      <h3>
        Stats
      </h3>
    </div>
    <div
      className="row"
    >
      <ul
        style={
          Object {
            "listStyleType": "none",
          }
        }
      >
        <li>
          Stars: 
          1
        </li>
        <li>
          Watchers: 
          2
        </li>
        <li>
          Forks: 
          3
        </li>
      </ul>
    </div>
  </div>,
  <div
    className="container"
  >
    <div
      className="row"
    >
      <h3>
        Languages
      </h3>
    </div>
    <div
      className="row"
    >
      <ul
        style={
          Object {
            "listStyleType": "none",
          }
        }
      >
        <li>
          Ruby
          : 
          4
        </li>
      </ul>
    </div>
  </div>,
]
`;
